src/sidebar/components/test/stream-search-input-test.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { shallow } = require('enzyme');
+const { createElement } = require('preact');
+const { act } = require('preact/test-utils');
+
+const StreamSearchInput = require('../stream-search-input');
+
+describe('StreamSearchInput', () => {
+  let fakeLocation;
+  let fakeRootScope;
+
+  beforeEach(() => {
+    fakeLocation = {
+      path: sinon.stub().returnsThis(),
+      search: sinon.stub().returns({ q: 'the-query' }),
+    };
+    fakeRootScope = {
+      $apply: callback => callback(),
+      $on: sinon.stub(),
+    };
+  });
+
+  function createSearchInput(props = {}) {
+    return shallow(
+      <StreamSearchInput
+        $location={fakeLocation}
+        $rootScope={fakeRootScope}
+        {...props}
+      />
+    ).dive(); // Dive through `withServices` wrapper.
+  }
+
+  it('displays current "q" search param', () => {
+    const wrapper = createSearchInput();
+    assert.equal(wrapper.find('SearchInput').prop('query'), 'the-query');
+  });
+
+  it('sets path and query when user searches', () => {
+    const wrapper = createSearchInput();
+    act(() => {
+      wrapper
+        .find('SearchInput')
+        .props()
+        .onSearch('new-query');
+    });
+    assert.calledWith(fakeLocation.path, '/stream');
+    assert.calledWith(fakeLocation.search, { q: 'new-query' });
+  });
+
+  it('updates query when changed in URL', () => {
+    fakeLocation.search.returns({ q: 'query-b' });
+    const wrapper = createSearchInput();
+
+    assert.calledOnce(fakeRootScope.$on);
+    assert.calledWith(fakeRootScope.$on, '$locationChangeSuccess');
+
+    act(() => {
+      fakeRootScope.$on.lastCall.callback();
```

Comment:
I've never encountered `sinon.call.callback` before...interesting! On the other hand...I had to look this up; I'm wondering if a comment might be in order here?

Reply:
I honestly don't see a way to add a whole lot of value here. In other words, I'm fine if the next person to read this just has to look it up as well ;)